go/pkg/credshelper/credshelper.go

```diff
@@ -209,53 +240,50 @@ func runCredsHelperCmd(credsHelperCmd *reusableCmd) (*oauth2.Token, time.Time, e
 		log.Errorf("Credentials helper warnings and errors: %v", stderr.String())
 	}
 	if err != nil {
-		return nil, time.Time{}, err
+		return credshelperOutput{}, err
 	}
-	token, expiry, refreshExpiry, err := parseTokenExpiryFromOutput(out)
-	return &oauth2.Token{
-		AccessToken: token,
-		Expiry:      expiry,
-	}, refreshExpiry, err
+	return parseTokenExpiryFromOutput(out)
 }
 
-// CredsHelperOut is the struct to record the json output from the credshelper.
-type CredsHelperOut struct {
-	Token         string `json:"token"`
-	Expiry        string `json:"expiry"`
-	RefreshExpiry string `json:"refresh_expiry"`
+// JSONOut is the struct to record the json output from the credshelper.
+type JSONOut struct {
+	Token         string            `json:"token"`
+	Headers       map[string]string `json:"headers"`
+	Expiry        string            `json:"expiry"`
+	RefreshExpiry string            `json:"refresh_expiry"`
 }
 
-func parseTokenExpiryFromOutput(out string) (string, time.Time, time.Time, error) {
+func parseTokenExpiryFromOutput(out string) (credshelperOutput, error) {
```

Comment:
You can return a pointer to credsHelperOutput and avoid unnecessary copies, while also returning `nil` in case of errors.

I don't mind the copy too much since we call creds helper sparsely, but it would be ideal to return a `nil` pointer in case of errors than an empty struct.

Reply:
Done